File: leetcode/MaxArea.scala
object Solution {
    def whileLoop(left : Int, 
                  right : Int, 
                  height : Array[Int],
                  maxArea : Int) : Int = {
        
        if (left >= height.length) {
            return maxArea;
        }

        
        var tempArea : Int = (maxArea).max(height(left).min(height(right)) * (right - left)); 
        
        if (height(right) >= height(left)) {
            return whileLoop(left + 1, right, height, tempArea);
        } else {
            return whileLoop(left, right  - 1, height, tempArea);
        }
        
        if (left.compare(right)  == 0){
            return tempArea;
        } else {
            0
        }
        

        
    }
   
    def maxArea(height: Array[Int]): Int = {
        
        if (height.length < 2) {
            return 0;
        }
        
        var maxArea : Int = 0;
        var left: Int  = 0; 
        var right: Int = height.length - 1;
        
        return whileLoop(left, right, height, maxArea);
        
    }
}

File: leetcode/maxArea.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    
    if (height.length < 2) {
        return 0;
    }
    
    let left = 0;
    let right = height.length - 1;
    let maxArea = 0; 
    
    while(true) {
        maxArea = Math.max(maxArea, Math.min(height[right], height[left]) * (right - left)); 
        
        if (height[right] >= height[left]) {
            left++;
        } else {
            right--;
        }
        
        if (left == right) {
            break;
        }
    }
    return maxArea;
    
};
